stl/src/excptptr.cpp

```diff
@@ -37,13 +37,37 @@ extern "C" _CRTIMP2 void* __cdecl __AdjustPointer(void*, const PMD&); // defined
 using namespace std;
 
 namespace {
-#ifdef _M_CEE_PURE
+#if defined(_M_CEE_PURE)
     template <class _Ty>
     _Ty& _Immortalize() { // return a reference to an object that will live forever
         /* MAGIC */ static _Immortalizer_impl<_Ty> _Static;
         return reinterpret_cast<_Ty&>(_Static._Storage);
     }
-#else // ^^^ _M_CEE_PURE ^^^ // vvv !_M_CEE_PURE vvv
+#elif _MSC_VER > 1927 && !defined(_M_CEE) // _M_CEE test is TRANSITION, VSO-1153256
+    template <class _Ty>
+    struct _Constexpr_excptptr_immortalize_impl {
+        union {
+            _Ty _Storage;
+        };
+
+        constexpr _Constexpr_excptptr_immortalize_impl() : _Storage() {}

```

Comment:
Should this be `noexcept`, and (stylistic nitpick) should it initialize `_Storage` with braces to imitate `_Constexpr_immortalize_impl` in the header?